docs/adr/0054-arborescence-apps-front.md

```diff
@@ -0,0 +1,76 @@
+# 54. Arborescence Apps Front
+
+Date : 2023-09-26
+
+## État
+
+En cours
+
+## Contexte
+
+### Constats
+
+- Lors de l'Onboarding, il faut du temps pour appréhender l’arborescence car différente de tout autre framework JavaScript.
+- La taille des Apps Front Pix ne fait que croître cependant, plus notre codebase grandit plus la question de l'arborescence est importante car il est de plus en plus complexe de retrouver les fichiers liés à une fonctionnalité.
+- Sans plugins ou coup de pouce, il est très difficile de retrouver les tests et le style d'un template rapidement
+
+### Problématique
+
+Repenser l’arborescence des apps front Pix afin de faciliter le travail des équipes et leur navigation à travers l'arborescence
+Éclaircir et documenter l’arborescence de Pix Orga pour commencer
+
+### Objectifs
+
+Structure claire et lisible pour les équipes
+Clarifier les intentions dès les premiers répertoires (et pouvoir approfondir pour détailler par couche)
+Limiter le nombre de fichiers par répertoire
+
+### Contraintes
+
+- Refactorings uniquement structurels
+- Changements iso-fonctionnels
+- Pas de régression (CI passe, tests passent, iso-perfs, pas de bug)
+- Ne pas dégrader l’expérience de développement sur Pix Orga
+- Avoir une arborescence qui puisse être appliquée sur les autres fronts de Pix
+
+## Solution 1 : Arborescence classique d'Ember
+
+### Avantages
+
+- Pas de refacto à réaliser
+- Directement ISO avec les autres fronts de PIX
+
+### Inconvénients
+
+- Complexité à retrouver le template lié au controller
+- Complexité à retrouver les tests liés au développement
+- Complexité à retrouver le style lié au développement
+
+## Solution 2 : Arborescence pods d'Ember
+
+### Avantages
+
+- Facile de s'y retrouver
+- Cohérence entre API et Fronts
+- Arborescence déjà mise en place chez Pix 1D
+- Ressemble plus aux autres frameworks JS
+
+### Inconvénients :
+
+- L'arborescence pods est bien supportée mais pas mise en avant par Ember.
```

Comment:
C'est bon signe ça ? Est-qu'on creuserai pas un peu dans les RFC Ember savoir si c'est pérène ?

Reply:
Il y a une issue où justement il parle du support de pods et la conclusion c'est qu'ils maintiennent. Je retrouve le lien et je le poste ici 😄

Reply:
Il y a des discussions dans ces deux issues : https://github.com/emberjs/rfcs/issues/650, https://github.com/emberjs/rfcs/issues/651

Concrètement, ce que je retiens c'est que ils ne lâcheront pas le support sans une alternative viable par peur de réduire l'adoption de la version qui introduirait ce breaking change. Idéalement ils aimeraient que ça sorte du code de ember-cli pour être dans une lib à côté mais les échanges n'ont pas évolué depuis 3 ans et je n'ai pas entendu dans les conférences sur Polaris parler du sujet.

Personnellement ça me suffit pour avoir confiance puisqu'ils ne droperont pas le support sans alternative viable.

Reply:
Je vais ajouter les deux liens dans l'ADR  ^^

Reply:
@yannbertrand a trouvé un nouveau lien à ce sujet : https://github.com/emberjs/rfcs/issues/906

Reply:
De ma lecture de cette issue https://github.com/emberjs/rfcs/issues/906, rien de nouveau sous le soleil : 
- Les pods intéressent particulièrement les grosses applications là où les plus petites restent plutôt dans la structure classic
- Pour des raisons, d'adoption avant de déprécié les pods il faut qu'une alternative existe qui incitera les grosses applications a continuer sur Ember
En résumé pas d'informations nouvelles par rapport aux liens listés ici à mon sens https://github.com/1024pix/pix/pull/7200#discussion_r1348369031

Reply:
@yannbertrand a posé la question du futurs des pods Ember sur Twitter et a obtenu une réponse :
https://x.com/nullvoxpopuli/status/1711735226282475636

Reply:
☝️ Pour pas avoir à cliquer : 

Question de Yann : 

> Is it planned to stop support for pods? What's the plan for Ember to avoid separation-by-type for big apps?

Réponse de NullVoxPopuli : 

> Gjs is the answer to everything.

Support for pods likely isn't going to be removed from ember-cli and embroider, but new features will replace the need for them.
Like how colorated components did for component pods.

Pour info Gjs, c'est expliqué ici sur notre Confluence : https://1024pix.atlassian.net/wiki/spaces/prescription/pages/3883696130/Lecture+RFC+first-class-component-templates

Reply:
Je note pour le moment :
Ajouter le lien https://github.com/emberjs/rfcs/issues/906 à la PR

Reply:
Ça fait un peu flipper de se dire qu'on part sur une arborescence qui n'est pas vraiment officielle, et qui sera remplacé par autre chose un de ces jours...

Et en même temps, c'est quand même rassurant de voir qu'il n'y aura pas de dépréciation avant d'avoir une autre option. J'imagine que la direction sera « similaire ». 
Ça signifie malgré tout qu'il faudra faire une autre migration sans doute. On peut espérer que l'esprit « clean archi » restera...